xray/src/event_loop.rs

```diff
@@ -55,7 +55,7 @@ impl EventLoop {
         loop {
             tokio::select! {
                 _ = &mut finish_timeout, if self.is_done => {
-                    self.on_finished(self.packets.len()).await?;
+                    self.on_finished(self.recv_counter).await?;
```

Comment:
This was printing that all packets were received even if none were, since `self.packets` contains all sent packets instead of all received packets (receiving a packet will just update the packet in the vector since it's basically storing metrics)